lib/model/recent_dm_conversations.dart

```diff
@@ -40,6 +42,24 @@ class RecentDmConversationsView extends ChangeNotifier {
 
   final int selfUserId;
 
+  // The ID of the latest messages exchanged with other users.
+  final Map<int, int> _dmRecencyData = {};
+
+  /// The ID of the latest message exchanged with this user.
+  ///
+  /// Returns -1 if there has been no DM message exchanged ever.
+  int getRecencyIndex(final int userId) => _dmRecencyData[userId] ?? -1;
+
+  void _computeDmRecencyData() {
+    for (final dmNarrow in sorted) {
+      for (final userId in dmNarrow.otherRecipientIds) {
+        if (!_dmRecencyData.containsKey(userId)) {
+          _dmRecencyData[userId] = map[dmNarrow]!;
+        }
+      }
+    }
+  }
```

Comment:
As for my understanding of `sorted`, I think it is the right direction. If a user is in multiple conversations, we only pick the latest message ID of the first-occurring conversation in `sorted` as it is the most recent one, all of the upcoming conversations for that user in `sorted` are later than that.

Reply:
Yeah, I believe this code's behavior is the right way around.

But this is a perfect example of something that would be good to cover with a test :slightly_smiling_face: — there should be a test where it's easy for the reader of the test to see that the test would fail if the implementation had this bug.